tests/test_config_consistency.py

```diff
@@ -237,3 +237,41 @@ def test_default_service_name(self):
         assert (
             spans[0]["service"] != "service_test"
         )  # in default scenario, DD_SERVICE is set to "weblog" in the dockerfile; this is a temp fix to test that it is not the value we manually set in the specific scenario
+
+
+@scenarios.tracing_config_nondefault
+@features.tracing_configuration_consistency
+class Test_Config_IntegrationEnabled_False:
+    """ Verify behavior of integrations automatic spans """
+
+    def setup_mongodb_integration_enabled_false(self):
+        self.r = weblog.get("/mongodb/getdatabase")
+
+    def test_mongodb_integration_enabled_false(self):
+        assert self.r.status_code == 200
+
+        mongodbSpans = []
+        # We do not use get_spans: the span we look for is not directly the span that carry the request information
+        for data, trace in interfaces.library.get_traces(request=self.r):
+            mongodbSpans += [(data, span) for span in trace if span.get("name") == "mongodb.query"]
+
+        assert len(mongodbSpans) == 0
+
+
+@scenarios.tracing_config_nondefault_3
+@features.tracing_configuration_consistency
+class Test_Config_IntegrationEnabled_True:
+    """ Verify behavior of integrations automatic spans """
+
+    def setup_mongodb_integration_enabled_true(self):
+        self.r = weblog.get("/mongodb/getdatabase")
+
+    def test_mongodb_integration_enabled_true(self):
+        assert self.r.status_code == 200
+
+        mongodbSpans = []
+        # We do not use get_spans: the span we look for is not directly the span that carry the request information
+        for data, trace in interfaces.library.get_traces(request=self.r):
+            mongodbSpans += [(data, span) for span in trace if span.get("name") == "mongodb.query"]
+
+        assert len(mongodbSpans) >= 1
```

Comment:
same here, we should assert that non mongodb spans are created

Reply:
Added here: https://github.com/DataDog/system-tests/pull/3060/commits/5c5392d93919668df67e9ff4e921ff22aad14395